magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleState.kt

```diff
@@ -6,27 +6,27 @@ import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.BACKWARDS
 import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.FORWARD
 import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.NO_MOVEMENT
 
-sealed class LifecycleState(open val context: Context? = null, private val order: Int) {
+public sealed class LifecycleState(public open val context: Context? = null, private val order: Int) {
 
   @SuppressLint("StaticFieldLeak")
-  object Destroyed : LifecycleState(null, 0)
-  data class Created(override val context: Context) : LifecycleState(context, 1)
-  data class Shown(override val context: Context) : LifecycleState(context, 2)
-  data class Resumed(override val context: Context) : LifecycleState(context, 3)
+  public object Destroyed : LifecycleState(null, 0)
+  public data class Created(override val context: Context) : LifecycleState(context, 1)
+  public data class Shown(override val context: Context) : LifecycleState(context, 2)
+  public data class Resumed(override val context: Context) : LifecycleState(context, 3)
 
-  fun getTheDirectionIShouldGoToGetTo(other: LifecycleState) = when {
+  internal fun getTheDirectionIShouldGoToGetTo(other: LifecycleState) = when {
     order > other.order -> BACKWARDS
     order == other.order -> NO_MOVEMENT
     order < other.order -> FORWARD
     else -> throw IllegalStateException("Unhandled order comparison: this is $order and other is ${other.order}")
   }
 
-  fun getEarlierOfCurrentState(): LifecycleState = when (this) {
+  internal fun getEarlierOfCurrentState(): LifecycleState = when (this) {
     is Destroyed -> Destroyed
     is Created, is Shown, is Resumed -> Created(context!!)
   }
 }
 
-enum class LifecycleStateDirection {
+internal enum class LifecycleStateDirection {
   FORWARD, BACKWARDS, NO_MOVEMENT;
 }
```

Comment:
I forget where we use these, but would we need these public to reimplement `LinearNavigator`? If so, we should leave them public

Reply:
d.